Update grid overlay to add tilings

Avengium opened the pull request:
# Description

Hi, I added several types of grid overlay to layer > grid.
The types are:
square rotated 45 degrees,
truncated square tiling,
tetrakis square,
triangular tiling,
trihexagonal tiling and
rhombille tiling with rhombus.
Inspired by these uniform tilings: https://en.wikipedia.org/wiki/List_of_Euclidean_uniform_tilings

# Type of change

<!-- Please put X into brackers of required option OR delete options that are not relevant -->

- [X] New feature
- [ ] Refactoring / style
- [ ] Documentation update / chore
- [ ] Other (please describe)

# Versioning

I didn't change any text on versioning.
The files have the same version, so you can decide the scale of the version and which number do you want to put as version.
If you want to merge this change with another ones to make a version with more changes, you can do that too.
Do as you see fit.

Azgaar commented:
Hm, I can understand triangle, but how others can be useful? I don't think anyone is using these other tilings in their games

Avengium commented:
Hi, i put these tilings on the map to allow people use and see these different tilings.
Is not specifically for games, but to allow them a way to imagine distances, routes between places, connections and proportions in the map and measure under different tiles. I think we should include these grids for these reasons alone, aside games.

These tilings are also used in games, both videogames and board games.

Tetrakis square and truncated square tiling (long list): 
https://boardgamegeek.com/geeklist/28732/games-played-alquerque-style-grid

Truncated square: 
https://boardgamegeek.com/boardgame/4099/keythedral
https://boardgamegeek.com/boardgame/963/gardens-alhambra
https://boardgamegeek.com/boardgame/3154/nautilus

Triangular tiling:
Colossal citadels (that's the name of the videogame): https://twitter.com/unevendungeon
Sylves (is on itchio too) https://www.boristhebrave.com/docs/sylves/1/
https://boardgamegeek.com/boardgame/16498/inka

Rhombus / rhomboid / rhombille grid:
https://boardgamegeek.com/boardgame/9341/lost-valley
Videogame Q*bert and clones: https://freeqbert.org/
rhombus tiling game (is not regular rhombus): https://github.com/davidquick/rhombus-tiling-visualiser

Azgaar commented:
Yes, it can be used in some abstract games, but I don't really see any real usage for the generated map. We can keep it, but would need to fix issues like that:

<img width="812" alt="image" src="https://github.com/Azgaar/Fantasy-Map-Generator/assets/26469650/7b8b9a0d-a46b-418f-9862-5c70d913c366">

Also would need to think what does "distance between cells" mean in this context:

<img width="1001" alt="image" src="https://github.com/Azgaar/Fantasy-Map-Generator/assets/26469650/085956f1-26b0-4bc8-9071-7d9fdba8bc57">

Avengium commented:
That finer line in rhombille is made using this code. `M 0 10 H 11 M 22 10 H 33`.
In a 33x20px rectangle. which means: move to x:0, y:10, draw a horizontal line towards x:11. And for the other line move to x:22, y:10 and draw a horizontal line towards x:33. Maybe the fact that i used H instead of L for these ones makes them have a different (thinner) stroke width.

Azgaar commented:
No, it's the fact that you have a viewport with stroke width not being considered. It should be, as well as bevel

Azgaar commented:
@Avengium, are you going to complete this?

Azgaar commented:
You need to fix the beforementioned issues

Avengium commented:
Could you explain to me in a way that helps me why those lines have different thicknesses and how to fix these different issues?

Azgaar commented:
> Could you explain to me in a way that helps me why those lines have different thicknesses and how to fix these different issues?

It looks a part of the stroke width is out of the viewbox and getting cut. Also I still don't think this tilling is useful.

Avengium commented:
Fixed stroke width of line in style grid for the added tilings in this request.

Avengium commented:
This pull is from a long ago. The commits that say "[Merge branch 'Azgaar:master' are no new features, only to update the code to the newest version.
What it needs to be accepted?
Should it be declined or accepted?

Avengium commented:
Even if the tiles are not regular on four sizes, are regular on 2 sides and are considered regular shape that fill the plane.
I will try to view in how many places is used the `scale` param and if i can put the info on these grids.

Azgaar commented:
There should some scheme showing the size visually for each grid. This can help.

Avengium commented:
SVG in the graphic map:
`<g id="gridOverlay" opacity="0.8" scale="1" dx="0" dy="0" type="pointyHex" stroke="#777777" stroke-width="0.5"><rect width="1707" height="791" fill="url(#pattern_pointyHex)" stroke="none"></rect></g>`

https://github.com/Azgaar/Fantasy-Map-Generator/blob/master/modules/ui/style.js#L191
`if (styleElement === "gridOverlay") {
`

Avengium commented:
Updated triangle overlay grid and rhombic, rhombille to use 25px and regular shape.
"pattern_triangle" width="25" height="43.3"
"pattern_rhombille" width="41.25" height="25"

Avengium commented:
Added an icon-info-circled on style → Grid → scale to help solve doubts.

Avengium commented:
Updated svg to adjust distance to center to center cell instead of distance = side.
Now the distance from the center of two adjacent cells that share a side is always distance.
Changed in square 45 degrees, triangular, rhombille.
 45 degrees.
Previous  was: "distance" 25 is two squares that share a vertex.
Current is: 25 is two squares that share a side.

in triangular, distance was 2 triangles that are horizontal, now is two triangles that share a side.

In trihexagonal grid, the distance is measured in pairs hexagon-hexagon and triangle-triangle. So is 25 too, or 75 when default is 1px = 3km.

Azgaar commented:
Looks good! Can you please add new ones to the wiki? That is the last thing

Avengium commented:
I added screenshots of grids on: https://github.com/Azgaar/Fantasy-Map-Generator/wiki/Scale-and-distance
explaining that the distance is the same on all grids. From adjacent cell centers.
